### PR TITLE
[WIP] Add new torchvision models

### PR DESCRIPTION
**Pull request by oke-aditya**
Closes https://github.com/pytorch/vision/issues/4289

Adds the following models.

- [x] MobileNetv3: - Mobilenet v3 small, Mobilenet v3 large
- [x] Mobilenet Segmentation: - lraspp_mobilenet_v3_large
- [x] DeepLab Models: - deeplabv3_mobilenet_v3_large deeplabv3_resnet50 
- [x] FCN: - fcn_resnet50,
- [x] MNasNet: - mnasnet0_5, mnasnet1_0
- [ ] EfficientNet: efficientnet_b0, efficientnet_b1, efficientnet_b2, efficientnet_b3, efficientnet_b4, efficientnet_b5, efficientnet_b6, efficientnet_b7

I modified the DeepLab and FCN files inplace to avoid duplicating. It needed a small rename in the file.
I have added a couple of images denoting architecture of mnasnet and mobilenetv3.
Let me know if those images are fine.
The model description is pending for lr_aspp, mnasnet and MobileNetV3.

But a small review at this point would be great.

cc @datumbox @NicolasHug

### CODE REVIEW

**Comment by oke-aditya** [comment]
The old pages are still persisting along with the new pages. But I don't rename the files they would convey a wrong meaning.

File: pytorch_vision_deeplabv3.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 layout: hub_detail
 background-class: hub-background
 body-class: hub
-title: Deeplabv3-ResNet101
-summary: DeepLabV3 model with a ResNet-101 backbone
+title: Deeplabv3
```
Comment (NicolasHug) [inline pytorch_vision_deeplabv3.md:5]: You did well changing the title here, however I would suggest not to change the file name. I agree with you that it somehow conveys the wrong meaning, but changing the file name will break the existing URL (strangely enough they seem to still exist, but it's even worse, as those URLs will never be properly updated).

That the URL doesn't fully reflect the page content isn't that big of an issue IMO, at least it's not as bad as breaking existing URLs.

I will update our runbook / review guidelines to indicate that we should try to have filenames that are as generic as possible, to avoid such issues in the future

File: pytorch_vision_deeplabv3.md [hunk 2]
```diff
@@ -86,14 +89,17 @@ plt.imshow(r)
 
 ### Model Description
 
-Deeplabv3-ResNet101 is constructed by a Deeplabv3 model with a ResNet-101 backbone.
+Deeplabv3-ResNet is constructed by a Deeplabv3 model using respective ResNet-50 or ResNet-101 backbone.
```
Comment (NicolasHug) [inline pytorch_vision_deeplabv3.md:92]: nit:

```suggestion
Deeplabv3-ResNet is constructed by a Deeplabv3 model using a ResNet-50 or ResNet-101 backbone.
```

File: pytorch_vision_fcn.md [hunk 3]
```diff
@@ -2,8 +2,8 @@
 layout: hub_detail
 background-class: hub-background
 body-class: hub
-title: FCN-ResNet101
-summary: Fully-Convolutional Network model with a ResNet-101 backbone
+title: FCN
```
Comment (NicolasHug) [inline pytorch_vision_fcn.md:5]: Here as well, let's please avoid breaking the existing URLs :)

File: pytorch_vision_fcn.md [hunk 4]
```diff
@@ -85,13 +87,14 @@ plt.imshow(r)
 
 ### Model Description
 
-FCN-ResNet101 is constructed by a Fully-Convolutional Network model with a ResNet-101 backbone.
+FCN-ResNet is constructed by a Fully-Convolutional Network model using respective ResNet-50 or ResNet-101 backbone.
```
Comment (NicolasHug) [inline pytorch_vision_fcn.md:90]:
```suggestion
FCN-ResNet is constructed by a Fully-Convolutional Network model using a ResNet-50 or ResNet-101 backbone.
```

**Review by NicolasHug — COMMENTED** [review]
Thanks a lot for the PR @oke-aditya , this is very useful

I took a brief  look even though this is still WIP, LMK when this is ready for a full round. Considering the big amount of changes I would suggest to include the rest of the models in other PRs, too keep reviews manageable.

Also, unfortunately our CI is completely toasted at the moment: https://app.circleci.com/pipelines/github/pytorch/hub/1244/workflows/a94283ed-e5cc-4c8c-b4cf-9ba78a03900c/jobs/1677 because of all those rate API limit issues. Hopefully it should be fixed in a near future, but sadly it means we'll have to wait until the CI is back again to merge this PR (but we can still do most of the review in the mean time)

File: pytorch_vision_deeplabv3.md (hunk 1 above)
Comment (datumbox) [reply pytorch_vision_deeplabv3.md:5]: Good catch. How about keeping the old url with a link to the new location? Is this possible given our setup?

Comment (NicolasHug) [reply pytorch_vision_deeplabv3.md:5]: I think we would need to keep the existing `pytorch_vision_deeplabv3_resnet101.md ` file, which is not ideal - it would also show-up needlessly in the hub search results, etc.

One option would be to have an HTTP redirection in place, perhaps with https://github.com/jekyll/jekyll-redirect-from

But I would keep that for further improvements and keep the same filenames for now :)

**Comment by oke-aditya** [comment]
Something I found out in log 

```
DEPRECATION NOTICE: Builds using the Trusty build image will fail after September 19, 2021

The build image for this site uses Ubuntu 14.04 Trusty Tahr, which is no longer supported.
All Netlify builds using the Trusty build image will begin failing in the week of September 19.

To avoid service disruption, please select a newer build image at the following link:
https://app.netlify.com/sites/pytorch-hub-preview/settings/deploys#build-image-selection

For more details, visit the build image migration guide:
https://answers.netlify.com/t/end-of-support-for-trusty-build-image-everything-you-need-to-know/39004
```

I think this should be addressed by pytorch.github.io team.

**Comment by NicolasHug** [comment]
>Something I found out in log

Thanks for raising this, we'll notify the relevant team

File: pytorch_vision_deeplabv3.md (hunk 1 above)
Comment (oke-aditya) [reply pytorch_vision_deeplabv3.md]: Great then, I have renamed the files. I think it works fine.

**Comment by oke-aditya** [comment]
Hmm the open in Colab will not work as of now. I think it would need the notebook in pytorch.gihtub.io repo.
Probably in [this](https://github.com/pytorch/pytorch.github.io/tree/site/assets/hub) folder ?

File: pytorch_vision_deeplabv3_resnet101.md [hunk 5]
```diff
@@ -35,7 +38,7 @@ So, `output['out']` is of shape `(N, 21, H, W)`. More documentation can be found
 ```python
 # Download an example image from the pytorch website
 import urllib
-url, filename = ("https://github.com/pytorch/hub/raw/master/images/dog.jpg", "dog.jpg")
+url, filename = ("https://github.com/pytorch/hub/raw/master/images/deeplab1.png", "deeplab1.png")
```
Comment (oke-aditya) [inline pytorch_vision_deeplabv3_resnet101.md:41]: We are actually showing image and output of this file but previously were demonstrating on dog.
I think we should demo on the same image.
This is also in Colab notebook, which would need update to pytorch.github.io repo.

I have rolled this change for all 3 segmentation model.s

File: pytorch_vision_deeplabv3_resnet101.md [hunk 6]
```diff
@@ -45,6 +48,7 @@ except: urllib.request.urlretrieve(url, filename)
 from PIL import Image
 from torchvision import transforms
 input_image = Image.open(filename)
+input_image = input_image.convert("RGB")
```
Comment (oke-aditya) [inline pytorch_vision_deeplabv3_resnet101.md:51]: Well that was a PNG image, we need a RGB one for transforms.

File: pytorch_vision_mobilenet_v3.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+|  Model structure          | Top-1 error | Top-5 error |
+|  ------------------       | ----------- | ----------- |
+|  mobilenet_v3_large       | 25.95       | 8.66        |
+|  mobilenet_v3_small       | 32.33       | 12.59       |
+
+
+### References
+
+ - [Searching for MobileNetV3](https://arxiv.org/abs/1905.02244)
```
Comment (oke-aditya) [inline pytorch_vision_mobilenet_v3.md:97]: Can we add link to the excellent blog created by Vasilis ?

Same for lrsapp. There are not many resources and the original paper has just 1 small paragraph.

https://pytorch.org/blog/torchvision-mobilenet-v3-implementation/

File: pytorch_vision_lraspp.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    output = model(input_batch)['out'][0]
+output_predictions = output.argmax(0)
+```
+
+The output here is of shape `(21, H, W)`, and at each location, there are unnormalized probabilities corresponding to the prediction of each class.
+To get the maximum prediction of each class, and then use it for a downstream task, you can do `output_predictions = output.argmax(0)`.
+
+Here's a small snippet that plots the predictions, with each color being assigned to each class (see the visualized image on the left).
+
+```python
```
Comment (oke-aditya) [inline pytorch_vision_lraspp.md:72]: Not sure if it would be good to use torchvision utils. 
Since they are from v0.10+ it might not be backward compatible.

Let me know.

**Review by oke-aditya — COMMENTED** [review]
Would love your thoughts @NicolasHug and @datumbox 

I need to go through the blog and papers to update descriptions of these models.

**Comment by NicolasHug** [comment]
> Something I found out in log
> 
> ```
> DEPRECATION NOTICE: Builds using the Trusty build image will fail after September 19, 2021
> 
> The build image for this site uses Ubuntu 14.04 Trusty Tahr, which is no longer supported.
> All Netlify builds using the Trusty build image will begin failing in the week of September 19.
> 
> To avoid service disruption, please select a newer build image at the following link:
> https://app.netlify.com/sites/pytorch-hub-preview/settings/deploys#build-image-selection
> 
> For more details, visit the build image migration guide:
> https://answers.netlify.com/t/end-of-support-for-trusty-build-image-everything-you-need-to-know/39004
> ```
> 
> I think this should be addressed by pytorch.github.io team.

@malfet @seemethere , @jspisak  suggested that one of you guys might be able to help with this :) ?
I checked and unfortunately I don't have the rights to update the ubuntu image myself.

File: pytorch_vision_mobilenet_v3.md (hunk 7 above)
Comment (datumbox) [reply pytorch_vision_mobilenet_v3.md:97]: Sure, we could add it as a reference if you think it's useful. One reason I did not add the article in the docstring is because I'm discussing closely the implementation at that point in time, which might be refactored and become obsolete. The blogpost does link to githash URLs so the references will be OK.

**Comment by oke-aditya** [comment]
I will complete the PR this week mostly. Sorry for the delay.

**Comment by datumbox** [comment]
@oke-aditya it seems to me you missed* one model.

\* Or I sneakily added the new EfficientNet your list. 😄

**Comment by oke-aditya** [comment]
So the only thing left now is to add descriptions about the models.

- [ ] Mnasnet
- [ ] EfficientNet
- [ ] MobileNetv3
- [ ] Lrsapp

I think I can finish this hopefully over the weekend.

File: pytorch_vision_efficientnet.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+github-id: pytorch/vision
+featured_image_1: efficientnet1.png
+featured_image_2: no-image
+accelerator: cuda-optional
+order: 10
+---
+
+```python
+import torch
+model = torch.hub.load('pytorch/vision:v0.11.0', 'efficientnet_b0', pretrained=True)
```
Comment (oke-aditya) [inline pytorch_vision_efficientnet.md:21]: I think that it would be wrong to mention `pytorch/vision:v0.10.0` as the model was not added till then.
It will be available in the coming 0.11 release.

Should we merge efficientnet now or after the 0.11 release?
I'm happy either ways, we can keep a  separate PR open and merge after the release.

Comment (datumbox) [reply pytorch_vision_efficientnet.md:21]: Yeah that's a good point... Good call. Do you want to remove this from the PR and send the addition on a separate? We could leave it unmerged until the release.

Comment (oke-aditya) [reply pytorch_vision_efficientnet.md:21]: Cool! I will remove the efficientnet files from current PR and keep them on separate one.
As you suggested, we can merge it after release :+1:

**Comment by NicolasHug** [comment]
@oke-aditya thanks for your work on this, in order to keep the reviewing manageable would you mind creating one PR per model?

**Comment by oke-aditya** [comment]
Ok sure no problem. I will close this PR and re-open new ones which add models one at a time.
